router/keeper/keeper.go

```diff
@@ -72,12 +98,36 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", "x/"+host.ModuleName+"-"+types.ModuleName)
 }
 
+func (k Keeper) WriteAcknowledgementForForwardedPacket(
```

Comment:
This is what gives us the ability to do async ack handling right?

Reply:
yes exactly. For the multi-hop refund cases, we avoid a local chain refund on the intermediate chains by calling this and NOT calling `OnAcknowledgementPacket`/`OnTimeoutPacket` which would issue a refund. This makes sure the funds burned/escrowed on the forward remain in that state so that funds are not duplicated.

Reply:
After this comment, I realized that we should:
- burn the escrowed funds on refund if the chain is the denom source for the forward channel but not the return channel
- transfer the funds from the forward channel escrow account to the return channel escrow account if the chain is the denom source for the forward channel and the return channel (denom is native denom on the chain)

With these updates, the escrow account balances are the same as if all of the transfers happened independently without the forward middleware.

I added escrow account assertions to the end to end tests and added a test to exercise both the escrow burn and escrow transfer: https://github.com/strangelove-ventures/ibctest/blob/393a8bc46994f5606e9f2fc669c7f6efe02eddf5/examples/ibc/packet_forward_test.go#L604-L616